portfolio/src/main/webapp/script.js

```diff
@@ -99,8 +99,10 @@ function fetchComments(quantity=5) {
               + authorEmailContent
               + '<p class="body-text">' + commentText + '</p></div>'; // outer <div> not closed yet
 
-          if (imageUrl == null) {
+          if (imageUrl === null) { // No image
             commentContent += '</div>'; // close outer div
+          } else if (imageUrl === undefined) { // Error occured
+            commentContent += '<div class="body-text">Error fetching image</div></div>';
```

Comment:
General comment: rather than handcrafting html with string manipulations, consider using `document.createElement()`, `element.appendChild()`, etc. to add elements to the page. No need to fix it here, but something to consider for the future.

Reply:
Yep, Eizan has also mentioned this to me previously, and I have tried it out for a previous version of the comments output, but it has so many more lines of code, so I thought it was overkill:

```
        var commentDiv = document.createElement('div'); // div to store one comment
        commentDiv.className = "comment";

        let commentText = commentData[i].commentText; 
        // Comment author name: "'Anonymous' if there was not a name submitted."
        let commentAuthor = commentData[i].commentAuthor === "" ? "Anonymous" : commentData[i].commentAuthor; 
        let imageUrl = commentData[i].attachedImage;

        let textContainer = document.createElement('div'); // Container for comment text & author name
        textContainer.className = "flex-item";
        
        // Author name <p> element
        let authorP = document.createElement('p');
        authorP.className = "body-text";
        let bolding = document.createElement('b');
        bolding.innerText = commentAuthor;
        authorP.appendChild(bolding);

        // Comment text <p> element
        let commentP = document.createElement('p');
        commentP.className = "body-text";
        commentP.innerText = commentText;

        textContainer.appendChild(authorP);
        textContainer.appendChild(commentP);
        commentDiv.appendChild(textContainer);

        // Display image as well, if there is an image
        if (imageUrl) {
          let imageContainer = document.createElement('div');
          imageContainer.className = "flex-item";
          
          let imageLink = document.createElement('a');
          imageLink.href = imageUrl;
          imageLink.target = "_blank"; // link will open in new tab
          
          let image = document.createElement('img');
          image.className = "comment-image";
          image.src = imageUrl;

          imageLink.appendChild(image);
          imageContainer.appendChild(imageLink);
          commentDiv.appendChild(imageContainer);
        }
        document.getElementById('comment-list').appendChild(commentDiv);
      }
```

I don't know if there is a way to shorten this? 😅

Reply:
Yeah ok, this is still hard to read haha.

The main things in the original code we want to do is limit string manipulations (e.g. try avoid all the `+=`s to append to the html string) and have all the HTML formatted and in as few assignments as possible so it's easy to see the structure. So, try do as much as possible in each innerHTML string assignment, and then edit it using element manipulation functions.

E.g below is a middle ground:

```js
let imageContent = "";
if (imageUrl === undefined) {
  imageContent = '<div class="body-text">Error fetching image</div>';
} else if (imageUrl !== null) {
  imageContent = `
    <div class="flex-item">
      <a href="${imageUrl}" target="_blank">
        <img class="comment-image" src="${imageUrl}">
      </a>
    </div>`;
}

var commentDiv = document.createElement('div'); // div to store one comment
commentDiv.className = "comment";
commentDiv.innerHTML = `
  <div class="flex-item">
    <p class="body-text"><b>${commentAuthor}</b></p>
    ${authorEmailContent}
    <p class="body-text">${commentText}</p>
  </div>
  ${imageContent}`
document.getElementById('comment-list').appendChild(commentDiv);
```

Although it still has some issues:
- User comments are inserted directly as html - generally you want to use `element.textContent = comment` to insert user data as they could contain html tags.
- I don't like how imageContent is done - I'd rather create it as an element and then `commentDiv.appendChild(imageContent)`

Reply:
Also long isn't necessarily bad, if making it longer makes it more readable, that's acceptable.